File: apps/node/test/dataset.test.ts
import { describe, expect, it } from "vitest";
import { dataDir } from "../src/dataset.ts";

describe("dataDir", () => {
  it("環境変数 ML_DATA_DIR が指定されていればそのディレクトリを返す", () => {
    const env: Record<string, string> = { ML_DATA_DIR: "/tmp/ml-data" };

    expect(dataDir((name) => env[name])).toBe("/tmp/ml-data");
  });

  it("環境変数が無ければ apps の data ディレクトリを返す", () => {
    expect(dataDir(() => undefined)).toBe("../data/sukkiri-ml");
  });
});

File: apps/ruby/test/dataset_test.rb
# frozen_string_literal: true

require "test_helper"

class DatasetTest < Minitest::Test
  def test_環境変数が無ければ既定の場所を使う
    assert_equal File.join("..", "data", "sukkiri-ml"), GettingStartedMl::Dataset.dir({})
  end

  def test_環境変数があればその場所を使う
    assert_equal "/tmp/data", GettingStartedMl::Dataset.dir({ "ML_DATA_DIR" => "/tmp/data" })
  end

  def test_環境変数が空なら既定の場所を使う
    assert_equal File.join("..", "data", "sukkiri-ml"), GettingStartedMl::Dataset.dir({ "ML_DATA_DIR" => "" })
  end
end
